model/browser/events.yaml

```
@@ -0,0 +1,100 @@
+groups:
+  - id: event.browser.resource_timing
+    stability: development
+    type: event
+    name: browser.resource_timing
+    brief: >
+      This event describes the timing metrics as provided by PerformanceResourceTiming Performance API.
+      These metrics are related to fetching a resource, such as
+      XMLHttpRequest, Fetch, sendBeacon APIs, SVG, image or script.
+    note: >
+      This event captures data from the [ResourceTiming](https://www.w3.org/TR/resource-timing/).
+      It is recommended to be captured by using the PerformanceResourceTiming API.
+      If the page starts unloading before the page loads,
+      then the partial data can be captured by calling the [performance.getEntriesByType](https://developer.mozilla.org/en-US/docs/Web/API/Performance/getEntriesByType) method.
+    attributes:
+      - ref: url.full
+        requirement_level: required
+        note: >
+          Generally the url doesn't include any query strings or any leading credentials,
+          and it's formatted as scheme://full domain/path only, for example https://user:pwd@www.example.com/path?a=b&c=d
+          would be sent as https://www.example.com/path by both removing any potential PII and reducing the cardinality of the name.
+    body:
+      id: browser.resource_timing
+      requirement_level: required
+      stability: development
+      type: map
+      fields:
+        - id: fetchStart
+          type: double
+          stability: development
+          brief: >
+            The time immediately before the browser starts to fetch the resource, in seconds.
```

I'd suggest just copying in the mozilla wording (and keeping it in milliseconds since that's how the underlying browser event is defined)
```suggestion
            A [timestamp](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp) immediately before the browser starts to fetch the resource.
```

---

Part of the problem with calling these "timestamps" (even though the standard calls these "DOMHighResTimeStamp") is that in practice that are not actually timestamps for the resources. they are a recorded number of milliseconds.

As per the spec, so these values fall into "the difference in time between two discrete points in time", where the timeOrigin can (and often is) zero.

> This type can be used to describe a discrete point in time or a time interval (the difference in time between two discrete points in time). The starting time can be either a specific time determined by the script for a site or app, or the [time origin](https://developer.mozilla.org/en-US/docs/Web/API/Performance/timeOrigin).

So I would prefer if this was either kept as `time` as it more accurately represents what it is (a duration of elapsed time not a specific timestamp)

---

@MSNev so this?

```
The [time](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp) immediately before the browser starts to fetch the resource.
```